src/sys/private/sockets_posix.nim

```diff
@@ -409,3 +419,78 @@ template tcpLocalEndpoint() {.dirty.} =
     result = IPEndpoint(kind: V6, v6: cast[IP6Endpoint](saddr))
   else:
     doAssert false, "Unexpected remote address family: " & $saddr.ss_family
+
+proc makeUnixSockaddr(path: string): Sockaddr_un =
+  result.sun_family = AF_UNIX.TSa_Family
+  if path.len >= Sockaddr_un_path_length:
+    raise newException(OSError, "socket path too long")
```

Comment:
I think `ValueError` is more appropriate here. `OSError` implies the existence of an OS error code embedded, for which we have none.

Reply:
We can preemptively raise `ENAMETOOLONG` here.